First release!

zhouyangyu opened the pull request:
Tests are made runnable on travis using pre-build docker images.

zhouyangyu commented:
🔥 I tested toil_strelka on LSF using singularity with real TGD data and it ran smoothly: 
`toil_strelka /ifs/work/leukgen/home/zhouy1/repos/toil_strelka_test/jobstore --stats --disableCaching --realTimeLogging --batchSystem LSF --writeLogs /ifs/work/leukgen/home/zhouy1/repos/toil_strelka_test/ --logFile /ifs/work/leukgen/home/zhouy1/repos/toil_strelka_test/head_job.toil --logLevel "INFO" --sequencing_method TGD --normal /ifs/res/leukgen/local/opt/leukdc/data/workflows/02/07/207/data/bam/I-H-100778-N1-1-D1-1.bam --tumor /ifs/res/leukgen/local/opt/leukdc/data/workflows/40/51/24051/data/bam/I-H-118643-T1-1-D1-1.bam --outdir /ifs/work/leukgen/home/zhouy1/repos/toil_strelka_test/ --reference /ifs/work/leukgen/ref/homo_sapiens/GRCh37d5/genome/gr37.fasta --bed /ifs/work/leukgen/home/zhouy1/repos/snv_benchmark/reference/IWG-PM.bed.gz --singularity /ifs/work/leukgen/home/zhouy1/repos/images/singularity/test-image-2018-03-14-f2580c7d1ef3.img --volumes /ifs /ifs`

juanesarango commented on toil_strelka/commands.py:
```diff
@@ -169,59 +176,16 @@ def get_parser():
     settings.add_argument(
-        "--manta",
-        help="path to manta folder",
+        "--sequencing_method",
```
You can add here:
`type=click.Choice(['WGS', 'WES', 'TGD'])`

juanesarango commented on toil_strelka/commands.py:
```diff
@@ -1,31 +1,33 @@
+            unitName (string): name of the job node.
+            args (list): positional arguments to be passed to `toil.job.Job`.
+            kwargs (dict): key word arguments to be passed to `toil.job.Job`.
+        """
```
This `self.options = options` is not necessary, because we have it from `ContainerJob` already.
So in fact the `BaseJob` Class is actually not necessary. Instead of using `BaseJob`, just use `ContainerJob`

jsmedmar replied:
I think joe may not need a `BaseJob` at all, just inherit from `ContainerJob`

juanesarango commented on run-test.sh:
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
```
Add this line to this file:
`set -euxo pipefail`

Apparently if your tests fail it won't fail in travis unless we trigger the fail with this command
see https://explainshell.com/explain?cmd=set+-euxo%20pipefail

zhouyangyu commented:
Made adjustment based on Juanes review. Added validation for essential programs if containers are not used, and warn user if those exe are not in the PATH. Updated README accordingly.

jsmedmar commented on toil_strelka/commands.py:
```diff
@@ -59,51 +60,56 @@ def run(self, fileStore):
         fileStore.logToMaster("Running strelka")
 
-        indel_candidates = os.path.join(self.manta_outdir, "results", "variants", "candidateSmallIndels.vcf.gz")
+        indel_candidates = os.path.join(self.manta_outdir,
```
be consistent with indentation, use something like:

```
indel_candidates = os.path.join(
    self.manta_outdir, "results",
    "variants", "candidateSmallIndels.vcf.gz"
    )
```

jsmedmar commented on toil_strelka/commands.py:
```diff
@@ -169,59 +176,16 @@ def get_parser():
-def process_parsed_options(options):
-    """Perform validations and add post parsing attributes to `options`."""
-    if options.writeLogs is not None:
-        subprocess.check_call(["mkdir", "-p", options.writeLogs])
```
You still want to create the logs directory, either in this function or in `main()`

jsmedmar commented on toil_strelka/commands.py:
```diff
@@ -112,10 +118,11 @@ def run(self, fileStore):
-    strelka_config_node = strelka_config(cores=1, memory="1G", options=options)
-    strelka_run_node = strelka_run(cores=1, memory="1G", options=options)
+    # memory requested here must be below the memory avaliable by docker
+    manta_config_node = MantaConfig(cores=1, memory="2G", options=options)
```
Do we have data to support these requirements?

zhouyangyu replied:
Those are the minimum requirements to run.

zhouyangyu commented:
Looks all good! 🎉
